tests/ssl-opt.sh

```diff
@@ -1620,7 +1620,7 @@ run_test() {
     fi
 
     if [ "$LIST_TESTS" -gt 0 ]; then
-        printf "%s\n" "$NAME"
+        printf "%s\n" "${TEST_SUITE_NAME:-ssl-opt};$NAME"
```

Comment:
Reporting `ssl-opt` as the test suite, rather than the individual file names from `opt-testcases`, is deliberate. Unlike individual `.data` files, the files in `opt-testcases` are not separate test suites that you can invoke directly. This previously came up in https://github.com/Mbed-TLS/mbedtls/pull/8088#discussion_r1317496917.

I'm not strongly opposed to using separate test suite names, but I don't see the point of changing it again. And I think if we do that we should make it possible to run them individually.

Reply:
> Reporting ssl-opt as the test suite, rather than the individual file names from opt-testcases, is deliberate.

I didn't notice that this is deliberate, but current things went wrong after that PR.

> Unlike individual .data files, the files in opt-testcases are not separate test suites that you can invoke directly.

Now we have `--test-suite` option in `ssl-opt.sh` to execute the "suite" seperately.

> I'm not strongly opposed to using separate test suite names.

Both are fine for me. Now the problem is the suite name we write to outcomes file and generated by `check_test_cases.py`. E.g.:

[CI reports](https://mbedtls.trustedfirmware.org/blue/rest/organizations/jenkins/pipelines/mbed-tls-pr-head/branches/PR-8561-head/runs/3/nodes/8063/log/?start=0) for #8561 :
```
[2023-11-28T22:16:33.848Z] Warning: Test case not executed: ssl-opt;TLS 1.3: m->G: all/all, good
```
This is false positive, because the suite name should be `tls13-kex-modes` in outcomes instead of `ssl-opt` generated by `check_test_cases.py`.

Reply:
> Now we have --test-suite option in ssl-opt.sh to execute the "suite" seperately.

Ah, good point. I'd forgotten we had this now.

What about test case duplication? One of the jobs of `check_test_cases` is to validate that if you take a fragment of the output of running one test suite, it's not ambiguous. If the same description is used in `opt-testcases/foo` and `opt-testcases/bar`:

* If we consistently report `ssl-opt` as the test suite, then `check_test_cases` will complain, so it's fine.
* If we consistently report the `opt-testcases` file name, then `check_test_cases` will check duplication inside each `opt-testcases` file, so there won't be ambiguity in the outcome file. However, the output from `ssl-opt.sh` itself will be ambiguous. Are we ok with that?